### PR TITLE
fix: align native chart interval and indicator popovers with design (OK-59326)

### PR DESCRIPTION
**Pull request by erikzou**
[OK-59326](https://onekeyhq.atlassian.net/browse/OK-59326)

----------

Hot-update fix targeting `release/v6.5.0`.

## Problem

Interval labels were truncated in the extension sidebar — `15分` / `30分` / `12时` rendered as `15…` / `30…` / `12…`.

Two causes stacked:

1. **Wrong type scale.** The pills used `$bodyLgMedium` (16px); the design specifies `bodyMd-Medium` (14px). The `adjustsFontSizeToFit` fallback that compensated for this is iOS-only, so the overflow only surfaced on web/extension.
2. **Double padding.** `Dialog`'s own `Content` already applies `px="$5" pb="$5"`, and `IntervalsDialogContent` added another `p="$5"` on top — 40px of horizontal padding instead of 20px. In the narrow sidebar that left each chip ~62px, just under what a 16px CJK label needs.

## Changes

**Typography and spacing, per design**
- Pill text `$bodyLgMedium` → `$bodyMdMedium`; removed `adjustsFontSizeToFit` / `minimumFontScale` so every chip renders at one size
- Pill: radius `$full` → `$2`, px `$3` → `$2.5`, grid gap `$2.5` → `$2`
- Pill colors now map to the design tokens: selected `$bgActive` + `$borderActive`, otherwise `$bgStrong`; label `$text` (was `$textSubdued` when inactive)
- Checkmark badge redrawn as the 13px corner tab the design specifies, with a 12px icon box
- Section headers → `$bodyMd`; `Edit` entry → design-system Button (small / tertiary, hover background suppressed)
- Dialog no longer double-pads (`contentContainerProps: { px: '$0', pb: '$0' }`); popover and dialog now carry their own padding and group gaps

**Behaviour**
- `All intervals` used the raw chart list while the edit page used a padded 14-item list, so the two disagreed. `getAllIntervalOptions` fabricated `disabled: true` placeholders for resolutions the chart never reported; it is now `getOrderedIntervalOptions` and only orders what the chart actually sent. Nothing the chart cannot render is offered anywhere.
- Toolbar segment items fill the full 30px so their active/hover background matches the adjacent `More` trigger (was 26px, which also squeezed the label)
- Intervals popover no longer moves focus into its content on open, which was giving the `Edit` button an unrequested focus ring
- Indicators popover opens with `bottom-start` like `More`, so it expands to the right instead of off the chart's left edge

**Shared styling**
- Pill layout, grid gap and color resolution move into `NativeChartControlsShared` so the interval and indicator grids cannot drift apart again

## Verification

`yarn agent:check --profile commit` passes; 9 TradingView test suites / 65 tests pass on this branch.

Not visually verified in a running build — the extension side panel cannot be driven from the local preview tooling. Spacing and type figures above were derived from the design tokens and checked against `tamagui.config.ts`.

[OK-59326]: https://onekeyhq.atlassian.net/browse/OK-59326?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/kit/src/components/TradingView/TradingViewV2/components/indicatorControls/NativeIndicatorControls.tsx [hunk 1]
```diff
@@ -109,13 +108,7 @@ function IndicatorPill({
...
       onPress={isDisabled ? undefined : onPress}
     >
-      <SizableText
-        size="$bodyMdMedium"
-        numberOfLines={1}
-        adjustsFontSizeToFit
-        minimumFontScale={0.82}
-        color={getIndicatorTextColor({ isActive, isDisabled })}
-      >
+      <SizableText size="$bodyMdMedium" numberOfLines={1} color={textColor}>
```
Comment (sidmorizon) [inline packages/kit/src/components/TradingView/TradingViewV2/components/indicatorControls/NativeIndicatorControls.tsx:111]: # [🟡 P2] 指标 pill 文字空间被压缩且同时移除了原生自动缩字，`StochRSI` 会被截断

## 问题

这一处同时发生了三个方向一致的收缩：共享的 `NATIVE_CHART_OPTION_PILL_LAYOUT_PROPS` 把指标 pill 的 `px` 从原来的 `$2`(8px) 提到 `$2.5`(10px)；`IndicatorListPopoverContent` 的容器内边距从 `p="$3"`(12px) 提到 `p="$5"`(20px)；同时这行把 `adjustsFontSizeToFit` / `minimumFontScale={0.82}` 删掉了，只留 `numberOfLines={1}`。

以 popover(`width: 360`，4 列，`gap $2`) 算：改前每格文字可用宽 `(360-2*12)/4 - 2*8 - 2 = 60px`，改后是 `(360-2*20)/4 - 2*10 - 2 = 52px`。而 `APP_NATIVE_INDICATOR_OPTIONS` 里最长的 `StochRSI` 在 `$bodyMdMedium`(14px, 500) 下约需 `~60px`。同一个 `IndicatorPill` 也被移动端 `IndicatorsDialogContent` 复用，在 375pt 机型上可用宽约 `55px`。

## 影响

主图/副图指标网格里 `StochRSI` 这类长标签会被省略成 `StochRS…`。`adjustsFontSizeToFit` 是 iOS/Android 专有属性，之前正是它在原生指标弹窗里兜住了这个 case（可缩到 `14*0.82≈11.5px`），删除后原生侧从「能缩字显示完整」直接退化成「截断」，而这个 dialog 并不是本次设计对齐的目标界面。Web 侧原本就贴边（60px vs 60px），本次再少 8px 后必然溢出。

## 建议

要么给指标 pill 保留原生自动缩字，要么让 pill 的水平内边距按网格来源区分（intervals 从 `$3`→`$2.5` 是变宽，indicators 从 `$2`→`$2.5` 是变窄，不适合共用一个常量）：

```ts
// NativeChartControlsShared.ts
export const NATIVE_CHART_OPTION_PILL_LAYOUT_PROPS = {
  // ...
  px: '$2',
} as const;
```

或在 `IndicatorPill` 上单独覆盖 `px="$2"` 并保留 `adjustsFontSizeToFit` / `minimumFontScale`，再用最长标签 `StochRSI` 在 375pt 机型和 360px popover 上各验一次。

[Claude session ↗](https://claude.ai/code/session_011GYGfzBzTeDPMU2w6S7JvV)

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by sidmorizon — COMMENTED** [review]
review

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/kit/src/components/TradingView/TradingViewV2/components/intervalSelector/NativeIntervalUtils.ts [hunk 2]
```diff
@@ -152,7 +144,14 @@ export function formatIntervalOptionDisplayLabel(
...
 
-export function getAllIntervalOptions(options: ITradingViewIntervalOption[]) {
+/**
+ * Orders the intervals the chart reported by the canonical template order and
+ * appends anything it reported outside that list. The chart owns the interval
+ * list, so a resolution it never reported is never surfaced.
+ */
+export function getOrderedIntervalOptions(
+  options: ITradingViewIntervalOption[],
+) {
```
Comment (sidmorizon) [inline packages/kit/src/components/TradingView/TradingViewV2/components/intervalSelector/NativeIntervalUtils.ts:154]: **Changed interval-list logic ships without any test coverage**

Severity: `non-severe`

The rewritten interval-ordering logic (`getOrderedIntervalOptions` at `packages/kit/src/components/TradingView/TradingViewV2/components/intervalSelector/NativeIntervalUtils.ts:152`) changes which time resolutions the chart offers, yet no test file covers this module, so the new "only offer what the chart reported" behavior is unverified.
Impact: A future regression in which intervals appear (or wrongly disappear) from the chart menus would go unnoticed.

<details>
<summary>Repository rule: modified logic requires accompanying tests</summary>

`.cursor/rules/self-testing.mdc` Section 6 mandates that new/modified logic has a corresponding `.test.ts` file covering happy path and edge cases, and that a bug fix includes a regression test. The behavior change here is not pure styling: `getAllIntervalOptions` previously fabricated `disabled: true` placeholders for all 14 template resolutions, whereas `getOrderedIntervalOptions` (`packages/kit/src/components/TradingView/TradingViewV2/components/intervalSelector/NativeIntervalUtils.ts:152-201`) now drops unmatched templates entirely, which changes `dialogOptions` in `packages/kit/src/components/TradingView/TradingViewV2/components/intervalSelector/hooks/useNativeIntervalSelector.ts:114-117` and therefore the edit grid, sorting order, and preferred-value reconciliation. There is no test file for `NativeIntervalUtils.ts` in the `intervalSelector` folder.

</details>

**Comment by sidmorizon** [comment]
@codex review

**Comment by sidmorizon** [comment]
@codex security review

**Comment by sidmorizon** [comment]
@cursoragent review

**Review by limichange — APPROVED** [review]

**Review by wabicai — APPROVED** [review]

**Review by sidmorizon — APPROVED** [review]
